internal/eval/partial_test.go

```diff
@@ -1238,6 +1238,30 @@ func TestPartialBasic(t *testing.T) {
 			nul,
 			testutil.Error,
 		},
+		{
+			"opGetTagKeep",
+			ast.Principal().GetTag(ast.String("key")),
+			ast.Principal().GetTag(ast.String("key")),
+			testutil.OK,
+		},
+		{
+			"opGetTagError",
+			ast.String("test").GetTag(ast.String("key")),
+			nul,
+			testutil.Error,
+		},
+		{
+			"opHasTagKeep",
+			ast.Principal().HasTag(ast.String("key")),
+			ast.Principal().HasTag(ast.String("key")),
+			testutil.OK,
+		},
+		{
```

Comment:
It would be nice to have folding examples here ... maybe there needs to be some extra data dumped into the context or something so that's possible? e.g. `context.entity` and then have some entity defined in the entities.